torch/csrc/jit/python/init.cpp

```diff
@@ -166,6 +166,8 @@ std::optional<IValue> toTypeInferredIValueOptional(py::handle input) {
 TORCH_API void runJITCPPTests();
 #endif
 
+static thread_local uint64_t cpp_to_python_translated_exception_count{0};
```

Comment:
My thought process here was:

(1) I figured this can't be a plain `uint64_t`, since multiple threads may be raising exceptions and can increment the counter.

(2) I originally made it a `std::atomic<uint64_t>`, but then I didn't want to risk the increment code being very slow, and exacerbating slowness problems in cases where there are many cpp-to-python exceptions. Alternatively, we could make it slow but gate it behind a config / env var, but this seems like a nice metric to be able to track in an always-on way.

(3) thread_local will under-count if multiple threads are all raising exceptions. My hope is that we are very unlikely to hit a situation where the main thread raises no cpp-to-python exception, but a side thread raises a large number (although if this happens, then... we will not be able to track it in metrics).

Reply:
Don't worry about (2), you're already in slow pain with exceptions, the atomic is a rounding error.